### PR TITLE
Issuance: c_nonce and jwt proof clarification

### LINKED ISSUE #188: c_nonce and jwt proof
in the issuance flow we use c_nonce and jwt proof without a proper definition of them

for instance, the c_nonce is a String containing a nonce to be used when creating a proof of possession of the key proof (see [Section 7.2](https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-wg-draft.html#credential_request)). When received, the Wallet MUST use this nonce value for its subsequent requests until the Credential Issuer provides a fresh nonce.

we have a quick definition here https://github.com/italia/eudi-wallet-it-docs/blob/versione-corrente/docs/en/pid-eaa-issuance.rst#token-response

we could imrpove this by giving more impl guidance about how this nonce is produces and how it can be used/verified with the jwt proof

### PR DESCRIPTION
**Pull request by fmarino-ipzs**
## Title
Issuance: c_nonce and jwt proof clarification

## Content
This PR closes #188 

## Review

- [x] Ensure your files are written following RST specs (*not MD!*)
- [ ] Italian version
- [x] English version
- [ ] Example files 
- [x] Ask for review

### CODE REVIEW

**Review by balanza — APPROVED** [review]
LGTM

File: docs/en/pid-eaa-issuance.rst [hunk 1]
```diff
@@ -308,7 +308,15 @@ The ``client_assertion`` is signed using the private key that is created during
...
+
+  1. all required claims for the *JWT proof* type are included as defined in the table of Section :ref:`Token Request <sec_token_request>`; 
+  2. the key proof is explicitly typed using header parameters as defined for that *proof* type; 
+  3. the header parameter indicates a registered asymmetric digital signature algorithm, alg parameter value MUST not be set to *none*; 
+  4. the signature on the key proof is verified with the public key contained in the header parameter;  
+  5. the header parameter does not contain a private key;
+  6. the *nonce* claim matches the server-provided *c_nonce* value, if the server had previously provided a *c_nonce*, the creation time of the JWT, as determined by either the issuance time (*iat*), or a server-managed timestamp via the *nonce* claim, is within an acceptable window.
```
Comment (peppelinux) [inline docs/en/pid-eaa-issuance.rst:318]: 1. the JWT proof MUST include all required claims as specified in the table of Section :ref:`Token Request <sec_token_request>`;
2. The key proof MUST be explicitly typed using header parameters as defined for the respective proof type;
3. The header parameter alg MUST indicate a registered asymmetric digital signature algorithm, and MUST NOT be set to `none`;
4. The signature on the key proof MUST be verified using the public key specified in the header parameter.
5. The header parameter MUST NOT contain a private key.
6. If a `c_nonce` value was previously provided by the server, the nonce claim in the JWT MUST match this `c_nonce` value. Furthermore, the creation time of the JWT, as indicated by the `iat` claim or a server-managed timestamp via the nonce claim, MUST be within an acceptable window of time as determined by the server.

File: docs/en/pid-eaa-issuance.rst [hunk 2]
```diff
@@ -308,7 +308,15 @@ The ``client_assertion`` is signed using the private key that is created during
...
 
-**Step 18 (Credential Request):** The Wallet Instance requests the Digital Credential to the PID/(Q)EAA Credential endpoint. The request MUST contain the *Access Token*, the *DPoP Proof JWT*, the *credential type*, the ``proof`` (proof of possession of the key) and the ``format`` parameters.
+**Step 18 (Credential Request):** The Wallet Instance requests the Digital Credential to the PID/(Q)EAA Credential endpoint. The request MUST contain the *Access Token*, the *DPoP Proof JWT*, the *credential type*, the `proof` (proof of possession of the key), and the ``format`` parameters, where the *proof* is an object containing the proof of possession of the cryptographic key material the issued PID/(Q)EAA Digital Credential would be bound to. The PID/(Q)EAA Provider performs the following checks on the Credential endpoint to verify the *proof*:
```
Comment (peppelinux) [inline docs/en/pid-eaa-issuance.rst]:
```suggestion
**Step 18 (Credential Request):** The Wallet Instance sends a request for the Digital Credential to the PID/(Q)EAA Credential endpoint. This request MUST include the Access Token, DPoP Proof JWT, credential type, proof (which demonstrates possession of the key), and format parameters. The proof parameter MUST be an object that contains evidence of possession of the cryptographic key material to which the issued PID/(Q)EAA Digital Credential will be bound. To verify the proof, the PID/(Q)EAA Provider conducts the following checks at the Credential endpoint:
```

**Review by peppelinux — CHANGES_REQUESTED** [review]
Editorials.

File: docs/en/pid-eaa-issuance.rst (hunk 1 above)
Comment (peppelinux) [inline docs/en/pid-eaa-issuance.rst]:
```suggestion
 1. the JWT proof MUST include all required claims as specified in the table of Section :ref:`Token Request <sec_token_request>`;
 2. The key proof MUST be explicitly typed using header parameters as defined for the respective proof type;
 3. The header parameter alg MUST indicate a registered asymmetric digital signature algorithm, and MUST NOT be set to `none`;
 4. The signature on the key proof MUST be verified using the public key specified in the header parameter.
 5. The header parameter MUST NOT contain a private key.
 6. If a `c_nonce` value was previously provided by the server, the nonce claim in the JWT MUST match this `c_nonce` value. Furthermore, the creation time of the JWT, as indicated by the `iat` claim or a server-managed timestamp via the nonce claim, MUST be within an acceptable window of time as determined by the server.
```

**Review by peppelinux — CHANGES_REQUESTED** [review]

**Review by peppelinux — APPROVED** [review]

**Pull request merged**